op-chain-ops/crossdomain/encoding.go

```diff
@@ -0,0 +1,72 @@
+package crossdomain
+
+import (
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	// NonceMask represents a mask used to extract version bytes from the nonce
+	NonceMask, _ = new(big.Int).SetString("0000ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", 16)
+	// relayMessage0ABI represents the v0 relay message encoding
+	relayMessage0ABI = "[{\"inputs\":[{\"internalType\":\"address\",\"name\":\"_target\",\"type\":\"address\"},{\"internalType\":\"address\",\"name\":\"_sender\",\"type\":\"address\"},{\"internalType\":\"bytes\",\"name\":\"_message\",\"type\":\"bytes\"},{\"internalType\":\"uint256\",\"name\":\"_messageNonce\",\"type\":\"uint256\"}],\"name\":\"relayMessage\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"}]"
+	// relayMessage1ABI represents the v1 relay message encoding
+	relayMessage1ABI = "[{\"inputs\":[{\"internalType\":\"uint256\",\"name\":\"_nonce\",\"type\":\"uint256\"},{\"internalType\":\"address\",\"name\":\"_sender\",\"type\":\"address\"},{\"internalType\":\"address\",\"name\":\"_target\",\"type\":\"address\"},{\"internalType\":\"uint256\",\"name\":\"_value\",\"type\":\"uint256\"},{\"internalType\":\"uint256\",\"name\":\"_minGasLimit\",\"type\":\"uint256\"},{\"internalType\":\"bytes\",\"name\":\"_message\",\"type\":\"bytes\"}],\"name\":\"relayMessage\",\"outputs\":[],\"stateMutability\":\"payable\",\"type\":\"function\"}]"
+	// relayMessage0 represents the ABI of relay message v0
+	relayMessage0 abi.ABI
+	// relayMessage1 represents the ABI of relay message v1
+	relayMessage1 abi.ABI
+)
+
+// Create the required ABIs
+func init() {
+	var err error
+	relayMessage0, err = abi.JSON(strings.NewReader(relayMessage0ABI))
+	if err != nil {
+		panic(err)
```

Comment:
Normally not a fan of panic, but here it will only fail if the ABI is wrong, so I'm OK.